### PR TITLE
Remove blockDownloadFailed event

### PR DESCRIPTION
**Pull request by earlbread**
This removes `blockDownloadFailed` event from `Swarm<T>.PreloadAsync()` and adds `BlockDownloadTimeout`.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -133,6 +133,8 @@ To be released.
      -  `BlockStates` (`0x23`)  [[#946]]
      -  `DifferentVersion` (`0x30`)  [[#949]]
  -  Every message now contains app protocol version in its header.  [[#949]]
+ -  Removed `PreloadBlockDownloadFailEventArgs` class.
+ -  Removed `blockDownloadFailed` parameter from `Swarm<T>.PreloadAsync()` method.
```
Comment (dahlia) [inline CHANGES.md:137]:
```suggestion
 -  Removed `PreloadBlockDownloadFailEventArgs` class.  [[#1002]]
 -  Removed `blockDownloadFailed` parameter from `Swarm<T>.PreloadAsync()` method.
     Use `SwarmOptions.BlockDownloadTimeout` instead.  [[#1002]]
```

Comment (riemannulus) [inline CHANGES.md:137]: We don't need an issue number here?
```suggestion
 -  Removed `PreloadBlockDownloadFailEventArgs` class. [[#1002]]
 -  Removed `blockDownloadFailed` parameter from `Swarm<T>.PreloadAsync()` method. [[#1002]]
```

Comment (riemannulus) [reply CHANGES.md:137]: nvm. same at https://github.com/planetarium/libplanet/pull/1002#discussion_r487644554

**Comment by limebell** [comment]
`BlockDownloadTimeout` seems fail 🤔

**Comment by dahlia** [comment]
@earlbread Could you rebase this?

**Comment by earlbread** [comment]
> @earlbread Could you rebase this?

I found the `BlockDownloadTimeout` is flaky. I will rebase after fixing it.

**Comment by dahlia** [comment]
@earlbread Seems update.  Is this ready to review?  Also, it needs to be rebased.

**Comment by earlbread** [comment]
I fixed the flaky test and rebased it, please review this again.

**Review by dahlia — APPROVED** [review]

**Review by longfin — APPROVED** [review]

**Review by limebell — APPROVED** [review]

**Pull request merged**